metals/src/main/scala/scala/meta/internal/metals/debug/SourcePathAdapter.scala

```diff
@@ -44,7 +44,7 @@ private[debug] final class SourcePathAdapter(
     sourceUri.getScheme match {
       case "jar" =>
         Option(AbsolutePath(Paths.get(sourceUri)).toFileOnDisk(workspace))
-      case "file" => Some(sourcePath.toAbsolutePath)
+      case "file" => Some(AbsolutePath(Paths.get(sourceUri)))
```

Comment:
Maybe extract `AbsolutePath(Paths.get(sourceUri))` into value as it's used in both cases?

Also, `toFileOnDisk`  might be used even for local files - in this case it won't create anything.